compiler/rustc_codegen_ssa/src/debuginfo/type_names.rs

```diff
@@ -50,7 +52,14 @@ fn push_debuginfo_type_name<'tcx>(
     qualified: bool,
     output: &mut String,
     visited: &mut FxHashSet<Ty<'tcx>>,
+    type_name_cache: &RefCell<FxHashMap<(Ty<'tcx>, bool), String>>,
 ) {
+    // Check if we have seen this type and qualifier before.
+    if let Some(type_name) = type_name_cache.borrow().get(&(&t, qualified)) {
+        output.push_str(&type_name.clone());
```

Comment:
This should be `output.push_str(&type_name[..]);`. No need to clone the string for accessing its contents.